src/NServiceBus.Persistence.CosmosDB/Saga/SagaPersister.cs

```diff
@@ -83,6 +83,11 @@ public Task Update(IContainSagaData sagaData, SynchronizedStorageSession session
                     }
                 }
 
+                if (sagaStream == null)
+                {
+                    return default;
```

Comment:
Should we throw here an unrecoverable exception instead so that messages get moved into the error queue? Because right now this would cause the saga not found handler to be triggered (let's say someone forgot to enable the migration mode)

Reply:
Just tested this and indeed if you forget to add the `EnableMigrationMode` then you get essentially message loss

```
2020-09-23 15:25:35.279 INFO  No saga found for timeout message f320df48-5909-4998-862c-ac3f00dc2eee, ignoring since the saga has been marked as complete before the timeout fired
2020-09-23 15:25:35.280 INFO  No saga found for timeout message 9a02a926-9ab4-4a9c-a5f9-ac3f00dc2ec1, ignoring since the saga has been marked as complete before the timeout fired
2020-09-23 15:25:35.279 INFO  No saga found for timeout message 9fef6914-2ea7-498b-bf55-ac3f00dc2ee5, ignoring since the saga has been marked as complete before the timeout fired
2020-09-23 15:25:35.280 INFO  No saga found for timeout message 2fd70b2a-a966-47da-9938-ac3f00dc2eba, ignoring since the saga has been marked as complete before the timeout fired
2020-09-23 15:25:35.280 INFO  No saga found for timeout message 6b2d5c81-e7e8-43f0-a6ec-ac3f00dc2ed8, ignoring since the saga has been marked as complete before the timeout fired
2020-09-23 15:25:35.279 INFO  No saga found for timeout message 61c9cbc1-2514-4cd1-897d-ac3f00dc2ec9, ignoring since the saga has been marked as complete before the timeout fired
2020-09-23 15:25:35.280 INFO  No saga found for timeout message 96c8d393-66c7-4c9a-888e-ac3f00dc2ede, ignoring since the saga has been marked as complete before the timeout fired
2020-09-23 15:25:35.279 INFO  No saga found for timeout message c5db1a5d-b5d5-4067-8ac3-ac3f00dc2eb3, ignoring since the saga has been marked as complete before the timeout fired
...
```

Reply:
I realized we cannot seperate the paths from a regular saga not found case so we cannot throw